src/overlay007/communication_club.c

```diff
@@ -0,0 +1,1731 @@
+    ldr r0, = commClubMan
+              mov r2, #3
+    ldr r1, [r0, #0]
+    ldr r0, [r1, #0x68]
+    add r1, #0x40
+    ldr r0, [r0, #8]
+    bl sub_0205D8F4
+ _0224A1E8:
+    ldr r0, = commClubMan
+              ldr r0, [r0, #0]
+    ldr r0, [r0, #0x68]
+    ldr r0, [r0, #0xc]
+    bl sub_02025E44
+    add r1, r0, #0
+    ldr r0, = commClubMan
+              ldr r0, [r0, #0]
+    add r0, #0x40
+    bl sub_0205D944
+    mov r0, #1
+    bl sub_02002AC8
+    mov r0, #0
+    bl sub_02002AE4
+    mov r0, #0
+    bl sub_02002B20
+    ldr r0, = commClubMan
+              mov r3, #0
+    ldr r2, [r0, #0]
+    mov r1, #1
+    str r3, [sp]
+    str r1, [sp, #4]
+    add r0, r2, #0
+    str r3, [sp, #8]
+    ldr r2, [r2, #0x14]
+    add r0, #0x40
+    bl PrintStringSimple
+    ldr r1, = commClubMan
+              ldr r1, [r1, #0]
+    add r1, #0x94
+    strb r0, [r1]
+ _0224A232:
+    ldr r0, = ov7_0224A34C
+              bl CommClubMan_SetTask
+ _0224A238:
+              add sp, #0xc
+    pop {r4, r5, pc}
+}
+
+static BOOL ov7_0224A244 (SysTask * param0, void * param1)
+{
+    CommClubManager * v0 = param1;
+
+    if (CommSys_CheckError()) {
+        CommClubMan_DestroyList(param0, v0);
+        CommClubMan_OpenMenu(0, 0);
+        ov7_0224B348(v0);
+    } else if (sub_020365A8(0, 3) == CommSys_CurNetId()) {
+        sub_020365D0();
+        CommClubMan_DestroyList(param0, v0);
+        ov7_0224B2DC(v0);
+    } else if (sub_020365A8(0, 5) != -1) {
+        sub_020365D0();
+        CommClubMan_DestroyList(param0, v0);
+        ov7_0224B2DC(v0);
+    } else if (sub_020365A8(0, 4) == CommSys_CurNetId()) {
+        CommClubMan_OpenMenu(97, 0);
+        CommClubMan_SetTask(CommClubTask_DifferentRegulation);
+    } else if (sub_020365A8(0, 2) == CommSys_CurNetId()) {
+        sub_020365D0();
+
+        v0->unk_90 = CommInfo_CountReceived();
+
+        if (!sub_0205DA04(commClubMan->unk_94)) {
+            sub_0201D730(commClubMan->unk_94);
+        }
+
+        CommClubMan_DestroyList(param0, v0);
+        ov7_0224B08C(v0);
+    } else if (ov7_0224B4E4()) {
+        CommClubMan_DestroyList(param0, v0);
+        ov7_0224B2DC(v0);
+    } else {
+        return 0;
+    }
+
+    return 1;
+}
+
+static void ov7_0224A34C (SysTask * param0, void * param1)
+{
+    CommClubManager * v0 = param1;
+
+    if (!sub_0205DA04(commClubMan->unk_94)) {
+        return;
+    }
+
+    if (ov7_0224A244(param0, param1)) {
+        (void)0;
+    } else if (( PAD_BUTTON_B )&gCoreSys.pressedKeys) {
+        CommClubMan_OpenMenu(3, 0);
+        CommClubMan_SetTask(ov7_0224A394);
+    }
+}
+
+static void ov7_0224A394 (SysTask * param0, void * param1)
+{
+    CommClubManager * v0 = param1;
+
+    if (ov7_0224A244(param0, param1)) {
+        (void)0;
+    } else if (sub_0205DA04(commClubMan->unk_94)) {
+        v0->unk_60 = sub_02002100(commClubMan->fieldSystem->unk_08, &Unk_ov7_0224ED0C, 1024 - (18 + 12) - 9, 11, 4);
+        CommClubMan_SetTask(ov7_0224A3E4);
+    }
```

Comment:
Does the following match? It feels more readable to me

```suggestion
    if (ov7_0224A244(param0, param1)) {
        return;
    }
    
    if (sub_0205DA04(commClubMan->unk_94)) {
        v0->unk_60 = sub_02002100(commClubMan->fieldSystem->unk_08, &Unk_ov7_0224ED0C, 1024 - (18 + 12) - 9, 11, 4);
        CommClubMan_SetTask(ov7_0224A3E4);
    }
```